cranelift/reader/src/run_command.rs

```diff
@@ -54,20 +54,11 @@ impl RunCommand {
         actual: &Vec<DataValue>,
         expected: &Vec<DataValue>,
     ) -> bool {
-        let are_equal = actual
-            .into_iter()
-            .zip(expected.into_iter())
-            .all(|(a, b)| match (a, b) {
-                // We need to bit compare the floats to ensure that we produce the correct values
-                // on NaN's. The test suite expects to assert the precise bit pattern on NaN's or
-                // works around it in the tests themselves.
-                (DataValue::F32(a), DataValue::F32(b)) => a.bits() == b.bits(),
-                (DataValue::F64(a), DataValue::F64(b)) => a.bits() == b.bits(),
-
-                // We don't need to worry about F32x4 / F64x2 Since we compare V128 which is already the
-                // raw bytes anyway
-                (a, b) => a == b,
-            });
+        let are_equal = actual.len() == expected.len()
+            && actual
+                .into_iter()
+                .zip(expected.into_iter())
+                .all(|(a, b)| a.bitwise_eq(b));
```

Comment:
For future reference, it's possible to write this more tersely. `Iterator::zip` will implicitly call `into_iter()` on whatever you pass to it, so you can just use `.zip(expected)`. And the implementation of `into_iter()` on `&Vec` is the same as calling `iter()` (they both return references to the elements of the vector) so you can use the shorter name. When I was playing with this locally I found that those two changes meant this part of the expression would fit on one line with `cargo fmt`s defaults. :grin: 

This is not remotely important, so I'm merging without that change and please don't bother writing another PR for it. This version is just fine as-is. But I thought I'd note the alternative.